### PR TITLE
BLD: Remove windows shim

### PR DESCRIPTION
**Pull request by jbrockmendel**
Just to see if this breaks the world.

### CODE REVIEW

File: pandas/_libs/src/headers/stdint.h [hunk 1]
```diff
@@ -1,10 +0,0 @@
-#ifndef _PANDAS_STDINT_H_
-#define _PANDAS_STDINT_H_
-
-#if defined(_MSC_VER) && (_MSC_VER < 1900)
```
Comment (lithomas1) [inline pandas/_libs/src/headers/stdint.h:4]: I guess the shim was only used for < VS 2015(https://dev.to/yumetodo/list-of-mscver-and-mscfullver-8nd).  We should be good here as our min. Visual Studio requirement should be VS2019.

**Review by lithomas1 — APPROVED** [review]
LGTM. Maybe someone more familiar with the C code should review though.

File: pandas/_libs/src/headers/stdint.h (hunk 1 above)
Comment (rhshadrach) [reply pandas/_libs/src/headers/stdint.h:4]: More of out curiosity than anything else, where do we specify the minimum version for VS?

Comment (jbrockmendel) [reply pandas/_libs/src/headers/stdint.h:4]: Absolutely no idea I opened cython/cython#2570 years ago in the hope this would become their problem

Comment (mroeschke) [reply pandas/_libs/src/headers/stdint.h:4]: > More of out curiosity than anything else, where do we specify the minimum version for VS?


There's a mention of VS2019 in our development environment docs

https://pandas.pydata.org/pandas-docs/stable/development/contributing_environment.html#installing-a-c-compiler

**Review by mroeschke — APPROVED** [review]

**Review by mroeschke — APPROVED** [review]
Just in case, I think it would be good to include a whatsnew note that a customs header for int types for VS 2019 compat has been removed.

**Comment by jbrockmendel** [comment]
> Just in case, I think it would be good to include a whatsnew note that a customs header for int types for VS 2019 compat has been removed.

I don't see how it's user-facing, but OK.  Suggested wording/location?

**Comment by mroeschke** [comment]
Okay on second thought, this should be okay. Thanks @jbrockmendel

**Comment by rhshadrach** [comment]
Looks like this also fixed the py38_32bit build.

**Pull request merged**